src/main/java/org/openmrs/addonindex/backend/Bintray.java

```diff
@@ -65,7 +69,28 @@ public AddOnInfoAndVersions getInfoAndVersionsFor(AddOnToIndex addOnToIndex) thr
 			return handlePackageJson(addOnToIndex, json);
 		}
 	}
-	
+
+	@Override
+	public Integer fetchDownloadCounts(AddOnToIndex toIndex, AddOnInfoAndVersions infoAndVersions) throws Exception {
+		//We won't be checking for existing info as download counts is dynamic
+		// and has to be fetched each time
+		logger.info("Fetching Download Counts for " + toIndex.getUid());
+		BintrayDownloadCounts downloadCounts;
+		Integer totalDownloadCounts = 0;
+		String json;
+		String downloadStatsUrl = downloadCountUrlFor(toIndex);
+		try {
+			json = restTemplateBuilder.build().getForObject(downloadStatsUrl, String.class);
+			downloadCounts = mapper.readValue(json, BintrayDownloadCounts.class);
```

Comment:
looking at this code again I see that this class should be called BintrayGeoStats, not BintrayDownloadCounts. Can you do a quick refactor->rename?